latch_cli/services/move.py

```diff
@@ -0,0 +1,112 @@
+import click
+import gql
+from gql.transport.exceptions import TransportQueryError
+
+from latch.gql._execute import execute
+from latch_cli.services.cp.ldata_utils import LDataNodeType, get_node_data
+from latch_cli.services.cp.path_utils import (
+    get_name_from_path,
+    get_path_error,
+    is_remote_path,
+)
+
+
+def move(
+    src: str,
+    dest: str,
+):
+    if not (is_remote_path(src) and is_remote_path(dest)):
+        raise ValueError(
+            f"`latch mv` cannot be used for local file operations. Please make sure"
+            f" both of your input paths are remote (beginning with `latch://`)"
+        )
+
+    acc_id, data = get_node_data(src, dest, allow_resolve_to_parent=True)
+
+    src_data = data[src]
+    dest_data = data[dest]
+
+    path_by_id = {v.id: k for k, v in data.items()}
+
+    if src_data.is_parent:
+        raise get_path_error(src, "not found", acc_id)
+
+    new_name = None
+    if dest_data.is_parent:
+        new_name = get_name_from_path(dest)
+    elif dest_data.type in {LDataNodeType.obj, LDataNodeType.link}:
+        raise get_path_error(dest, "object already exists at path.", acc_id)
+
+    success_msg = "Move Succeeded."
+
+    try:
+        execute(
+            gql.gql("""
+            mutation Move(
+                $argNode: BigInt!
+                $argDestParent: BigInt!
+                $argNewName: String
+            ) {
+                ldataMove(
+                    input: {
+                        argNode: $argNode
+                        argDestParent: $argDestParent
+                        argNewName: $argNewName
+                    }
+                ) {
+                    clientMutationId
+                }
+            }"""),
+            {
+                "argNode": src_data.id,
+                "argDestParent": dest_data.id,
+                "argNewName": new_name,
+            },
+        )
+    except TransportQueryError as e:
+        if e.errors is None or len(e.errors) == 0:
+            raise e
+
+        msg: str = e.errors[0]["message"]
+
+        if msg.startswith("Permission denied on node"):
```

Comment:
Ik this is not you but I hate that our plpgsql errors are verbose strings. Would be nice if they were stringified json with a particular format or just broader categories which we could conform to like "permission_or_dne" / "conflict" / "share_link"

Reply:
yea we should start using json strings